apps/system/test/apps/longlaunchapp/test.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+(function() {
+    var fakeData = null;
+
+    window.launchDuration = 2000;
+    window.startTime = new Date().getTime();
+
+    window.sleepOn = function() {
+      var script = document.createElement('script');
+      script.src = '/wait.js';
+      script.async = false;
+      document.head.appendChild(script);
+    };
+
+    function print(str) {
+      var p = document.createElement('p');
+      p.textContent = str;
+      document.body.appendChild(p);
+    }
+
+    window.sayDone = function() {
+      print('DONE!');
+    };
+
+    navigator.mozSetMessageHandler('notification', function(m) {
+      fakeData = m;
+      print('notification: ' + JSON.stringify(m));
+    });
+
+    window.getFakeData = function() {
```

Comment:
no need to call this `fakeData` anymore since you are using the real message handler for this. i think it would be more clear to call it something like 'lastNotificationData` or something